From b67c67483f8c71c27a2965c5379ab1b81f278527 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Wed, 3 Apr 2024 11:01:11 +0200
Subject: [PATCH 1079/1141] Revert part of "media: hantro: improve decoder
 robustness"

Revert removal of hardware timeout feature.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: I1b2c7bc26836c5ee01000a43e12e4d2f5ce6369f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/371362
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 drivers/media/platform/verisilicon/hantro_g1_h264_dec.c | 1 +
 drivers/media/platform/verisilicon/hantro_g1_vp8_dec.c  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/verisilicon/hantro_g1_h264_dec.c b/drivers/media/platform/verisilicon/hantro_g1_h264_dec.c
index cc06e36403b5..0ebb6f2df5c2 100644
--- a/drivers/media/platform/verisilicon/hantro_g1_h264_dec.c
+++ b/drivers/media/platform/verisilicon/hantro_g1_h264_dec.c
@@ -269,6 +269,7 @@ int hantro_g1_h264_dec_run(struct hantro_ctx *ctx)
 	/* Start decoding! */
 	vdpu_write_relaxed(vpu,
 			   G1_REG_CONFIG_DEC_AXI_RD_ID(0xffu) |
+			   G1_REG_CONFIG_DEC_TIMEOUT_E |
 			   G1_REG_CONFIG_DEC_OUT_ENDIAN |
 			   G1_REG_CONFIG_DEC_STRENDIAN_E |
 			   G1_REG_CONFIG_DEC_2CHAN_DIS |
diff --git a/drivers/media/platform/verisilicon/hantro_g1_vp8_dec.c b/drivers/media/platform/verisilicon/hantro_g1_vp8_dec.c
index a4e66d16ce3c..b12f090a0b25 100644
--- a/drivers/media/platform/verisilicon/hantro_g1_vp8_dec.c
+++ b/drivers/media/platform/verisilicon/hantro_g1_vp8_dec.c
@@ -452,7 +452,7 @@ int hantro_g1_vp8_dec_run(struct hantro_ctx *ctx)
 
 	hantro_vp8_prob_update(ctx, hdr);
 
-	reg = G1_REG_CONFIG_DEC_AXI_RD_ID(0xffu) |
+	reg = G1_REG_CONFIG_DEC_TIMEOUT_E |
 	      G1_REG_CONFIG_DEC_STRENDIAN_E |
 	      G1_REG_CONFIG_DEC_INSWAP32_E |
 	      G1_REG_CONFIG_DEC_STRSWAP32_E |
-- 
2.39.2

